Build fails on Apple platforms due to old zlib-1.2.13

Build fails on macOS/arm64 and macOS/am64 due to a failure in the dependent library zlib-1.2.13/zutil.h:
```
Allans-MBP2019:masorange_rules_helm allanc$ bazel build //...
INFO: Invocation ID: 7318cb30-8865-4788-98aa-d78597bb1454
INFO: Analyzed 268 targets (154 packages loaded, 3224 targets configured).
INFO: Found 268 targets...
ERROR: /private/var/tmp/_bazel_allanc/4c3b4cba7408dbec30df9531ae5d9289/external/zlib~1.2.13/BUILD.bazel:5:11: Compiling zutil.c [for tool] failed: (Exit 1): wrapped_clang failed: error executing command (from target @zlib~1.2.13//:zlib) external/bazel_tools~cc_configure_extension~local_config_cc/wrapped_clang '-D_FORTIFY_SOURCE=1' -fstack-protector -fcolor-diagnostics -Wall -Wthread-safety -Wself-assign -fno-omit-frame-pointer -g0 -O2 ... (remaining 36 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
In file included from external/zlib~1.2.13/zutil.c:8:
external/zlib~1.2.13/zutil.h:170:11: warning: 'OS_CODE' macro redefined [-Wmacro-redefined]
  170 | #  define OS_CODE 19
      |           ^
external/zlib~1.2.13/zutil.h:141:11: note: previous definition is here
  141 | #  define OS_CODE  7
      |           ^
In file included from external/zlib~1.2.13/zutil.c:10:
In file included from external/zlib~1.2.13/gzguts.h:21:
```

## Mitigation

Updating to a newer zlib mitigates (confirmed with zlib-1.3.1.bcr.5)

--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -11,6 +11,11 @@ bazel_dep(name = "bazel_skylib", version = "1.5.0")
 bazel_dep(name = "rules_pkg", version = "0.9.1")
 
 bazel_dep(name = "rules_oci", version = "1.6.0")
+
+# override zlib-1.2.13 in deps from @bazel_tools and @protobuf
+# may be removable if newer bazel versions carry in newer bazel_tools
+bazel_dep(name = "zlib", version = "1.3.1.bcr.5")
+
 bazel_dep(name = "stardoc", version = "0.6.2", dev_dependency = True)
 
 oci = use_extension("@rules_oci//oci:extensions.bzl", "oci", dev_dependency = True)

